test/test_timers.py

```diff
@@ -8,27 +7,33 @@
 from nameko.testing.utils import wait_for_call
 
 
-def test_provider():
+def test_provider(interval=0.1, eager=False):
     container = Mock(spec=ServiceContainer)
     container.service_name = "service"
     container.spawn_managed_thread = eventlet.spawn
 
-    timer = Timer(interval=0.1).bind(container, "method")
+    timer = Timer(interval, eager).bind(container, "method")
     timer.setup()
     timer.start()
 
-    assert timer.interval == 0.1
+    assert timer.interval == interval
+    assert timer.eager == eager
 
     with wait_for_call(1, container.spawn_worker) as spawn_worker:
-        with Timeout(1):
-            timer.stop()
+        timer.handle_result(None, None, None)
+        timer.stop()
 
     # the timer should have stopped and should only have spawned
     # a single worker
-    spawn_worker.assert_called_once_with(timer, (), {})
+    spawn_worker.assert_called_once_with(
+        timer, (), {}, handle_result=timer.handle_result)
     assert timer.gt.dead
 
```

Comment:
This change should be considered optional to for this PR. I'd prefer to land these fixes rather than force them to sit behind stylistic cleanup of old tests.